tests/test_orchestration_executor.py

```diff
@@ -228,6 +228,169 @@ def orchestrator(ctx: task.OrchestrationContext, orchestrator_input):
+    executor = worker._OrchestrationExecutor(registry, TEST_LOGGER)
+    actions = executor.execute(TEST_INSTANCE_ID, old_events, new_events)
+    assert len(actions) == 2
+    assert actions[1].HasField("scheduleTask")
+    assert actions[1].id == 1
+
+    # Simulate the task failing for the second time and confirm that a timer is scheduled for 2 seconds in the future
+    old_events = old_events + new_events
+    expected_fire_at = current_timestamp + timedelta(seconds=2)
+    new_events = [
+        helpers.new_orchestrator_started_event(current_timestamp),
+        helpers.new_task_failed_event(1, ValueError("Kah-BOOOOM!!!"))]
+    executor = worker._OrchestrationExecutor(registry, TEST_LOGGER)
+    actions = executor.execute(TEST_INSTANCE_ID, old_events, new_events)
+    assert len(actions) == 3
+    assert actions[2].HasField("createTimer")
+    assert actions[2].createTimer.fireAt.ToDatetime() == expected_fire_at
+    assert actions[2].id == 3
+
+    # Simulate the timer firing at the expected time and confirm that another activity task is scheduled
+    current_timestamp = expected_fire_at
+    old_events = old_events + new_events
+    new_events = [
+        helpers.new_orchestrator_started_event(current_timestamp),
+        helpers.new_timer_fired_event(3, current_timestamp)]
+    executor = worker._OrchestrationExecutor(registry, TEST_LOGGER)
+    actions = executor.execute(TEST_INSTANCE_ID, old_events, new_events)
+    assert len(actions) == 3
+    assert actions[1].HasField("scheduleTask")
+    assert actions[1].id == 1
+
+    # Simulate the task failing for a third time and confirm that a timer is scheduled for 4 seconds in the future
+    expected_fire_at = current_timestamp + timedelta(seconds=4)
+    old_events = old_events + new_events
+    new_events = [
+        helpers.new_orchestrator_started_event(current_timestamp),
+        helpers.new_task_failed_event(1, ValueError("Kah-BOOOOM!!!"))]
+    executor = worker._OrchestrationExecutor(registry, TEST_LOGGER)
+    actions = executor.execute(TEST_INSTANCE_ID, old_events, new_events)
+    assert len(actions) == 4
+    assert actions[3].HasField("createTimer")
+    assert actions[3].createTimer.fireAt.ToDatetime() == expected_fire_at
+    assert actions[3].id == 4
+
+    # Simulate the timer firing at the expected time and confirm that another activity task is scheduled
+    current_timestamp = expected_fire_at
+    old_events = old_events + new_events
+    new_events = [
+        helpers.new_orchestrator_started_event(current_timestamp),
+        helpers.new_timer_fired_event(4, current_timestamp)]
+    executor = worker._OrchestrationExecutor(registry, TEST_LOGGER)
+    actions = executor.execute(TEST_INSTANCE_ID, old_events, new_events)
+    assert len(actions) == 4
+    assert actions[1].HasField("scheduleTask")
+    assert actions[1].id == 1
+
+    # # Simulate the task failing for a fourth time and confirm that a timer is scheduled for 8 seconds in the future
```

Comment:
Why `# #` instead of just `#` for these comments?

Reply:
Updated